.githooks/pre-commit

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if which swiftformat >/dev/null; then
+	./.githooks/git-format-staged --formatter "swiftformat stdin --stdinpath '{}'" "apple/*.swift"
+else
+	echo "error: SwiftFormat not installed"
+	echo "error: Install it via 'brew install swiftformat'"
+	echo "error: or download from https://github.com/nicklockwood/SwiftFormat"
+	exit 1
```

Comment:
Would it be possible to either skip this check if no Swift changes are staged OR skip the check on non-Apple platforms? All the core devs here use macOS, but I know some end-users do occasionally develop on Windows or Linux if they are android-specific, and this would exclude them from being able to contribute patches easily.

Reply:
good point, I didn't consider non-Apple developers.